attestation-service/attestation-service/src/policy_engine/opa/mod.rs

```diff
@@ -161,6 +162,60 @@ impl PolicyEngine for OPA {
             .await
             .map_err(|e| anyhow!("Write OPA policy to file failed: {:?}", e))
     }
+
+    async fn remove_policies(&mut self, policy_id: String) -> Result<()> {
+        let policy_dir_path = self
+            .policy_dir_path
+            .to_str()
+            .ok_or_else(|| anyhow!("Miss Policy DirPath"))?;
+        let policy_file_path = format!("{policy_dir_path}/{policy_id}.rego");
+        tokio::fs::remove_file(policy_file_path)
+            .await
+            .context("Remove OPA policy file failed")?;
+
+        Ok(())
+    }
+
+    async fn list_policies(&self) -> Result<Vec<PolicyListEntry>> {
+        let mut policy_ids = Vec::new();
+        let policy_dir_path = self
+            .policy_dir_path
+            .to_str()
+            .ok_or_else(|| anyhow!("Miss Policy DirPath"))?;
+        let entries = std::fs::read_dir(policy_dir_path)?;
+        for entry in entries {
+            let entry = entry?;
+            let path = entry.path();
+
+            if path.extension().and_then(std::ffi::OsStr::to_str) == Some("rego") {
+                if let Some(filename) = path.file_stem() {
+                    if let Some(filename_str) = filename.to_str() {
+                        policy_ids.push(filename_str.to_owned());
+                    }
+                }
+            }
+        }
+
+        let mut policy_list = Vec::new();
+
+        for id in policy_ids.iter() {
+            let policy_file_path = format!("{policy_dir_path}/{id}.rego");
+            let policy = tokio::fs::read_to_string(policy_file_path)
+                .await
+                .context("Read OPA policy file failed")?;
+
+            let mut hasher = Sha384::new();
```

Comment:
is the use of this particular hash algo documented somewhere? It might be sufficient to add it as a comment to the API fn or PolicyListEntry struct